src/os/bluestore/BlueStore.cc

```diff
@@ -5129,7 +5129,10 @@ bool BlueStore::test_mount_in_use()
 int BlueStore::_minimal_open_bluefs(bool create)
 {
   int r;
-  bluefs = new BlueFS(cct);
+  if (cct->_conf->bluefs_alloc_size > 0)
```

Comment:
let's make this decision at BlueFS level and hence unconditionally pass min_alloc_size here. Just not to have bluefs logic outside of the class.
So just leave single BlueFS ctor which gets bluestore_min_alloc_size and passes it inside _default_bluefs_alloc_size where final decision is made. Or even simpler - just do the decision at ctor and remove _default_bluefs_alloc_size at all - it's used just once.

Reply:
That's what I wanted to do eventually, just wasn't sure what the best way to handle https://github.com/ceph/ceph/blob/master/src/os/bluestore/bluestore_tool.cc#L176 was?

Reply:
Actually the ideal case would be to use config parameter(s) on mkfs and then save applied value and reuse on subsequent openings. We have this approach for bluestore's min_alloc_size. Hence one is unable to alter the value and introduce any issues/incompatibilities this way. And this will also answer your question what value to use in bluestore_tool.

Reply:
I added a commit to address your comment. I don't think we strictly need to persist bluefs_alloc_size since 1. it is being derived from bluestore's min_alloc_size, which is already persisted 2. the plan going forward is to have just one alloc size, so we might not want to add more complexity to save bluefs_alloc_size and potentially complicate upgrades